apps/command-palette-wp-admin/src/use-commands.ts

```diff
@@ -0,0 +1,36 @@
+import { COMMANDS } from '@automattic/command-palette';
+import deepmerge from 'deepmerge';
+import type { Command, CommandCallBackParams } from '@automattic/command-palette';
+
+const waitForElementAndClick = ( selector: string, attempt = 1 ) => {
+	const element = document.querySelector< HTMLElement >( selector );
+	if ( element ) {
+		element.click();
+	} else if ( attempt <= 5 ) {
+		// Try again in 250ms, but no more than 5 times.
+		setTimeout( () => waitForElementAndClick( selector, attempt + 1 ), 250 );
+	}
+};
+
+export const useCommandsWpAdmin = (): Command[] => {
+	// Only override commands that need a specific behavior for WP Admin.
+	// Commands need to be defined in `packages/command-palette/src/commands.tsx`.
+	const commands = deepmerge( COMMANDS, {
```

Comment:
Why do we need to deepmerge? Couldn't we modify the commands by accessing the map/dictionary?

Reply:
It's not obvious here because we don't have so many commands to override, but for the Calypso commands (`wpcom-smp-commands.tsx`) it felt more readable to deepmerge than modifying multiple commands like this:

```
COMMANDS.getHelp.callback = () => {
  // ...
};
COMMANDS.clearCache.callback = () => {
  // ...
};
COMMANDS.enableEdgeCache.callback = () => {
  // ...
};
COMMANDS.disableEdgeCache.callback = () => {
  // ...
};
// ...
```

In the sense that you need to pay a little bit more of attention to see which commands are being modified because of the `COMMANDS.` prefix and `.callback` suffix being repeated everywhere.

Reply:
Could we remove it from here? That way we would reduce the package size by not needing the dependency on wp-admin